### PR TITLE
Issue 2580: Disable hostnamechecks for AutoScaleProcessor.

### LINKED ISSUE #2580: unable to setup trustmanager exception when TLS is enabled
**Problem description**

Enable TLS in config/standalone-config.properties and start standalone pravega.
The follow exception shows repeatedly in pravega log. Note that the pravega certificate was added to local java truststore.
```
2018-05-11 11:09:49,850 34706 [clientInternal-4] WARN  i.p.c.s.i.SegmentOutputStreamImpl - 98a4b7e5-56f8-4e04-a48b-fefb159615bf Failed to connect: 
java.util.concurrent.CompletionException: java.lang.RuntimeException: javax.net.ssl.SSLException: unable to setup trustmanager
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:961)
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:926)
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: javax.net.ssl.SSLException: unable to setup trustmanager
        at io.pravega.client.netty.impl.ConnectionFactoryImpl.establishConnection(ConnectionFactoryImpl.java:117)
        at io.pravega.client.segment.impl.SegmentOutputStreamImpl.lambda$null$1(SegmentOutputStreamImpl.java:522)
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:952)
        ... 9 common frames omitted
Caused by: javax.net.ssl.SSLException: unable to setup trustmanager
        at io.netty.handler.ssl.ReferenceCountedOpenSslClientContext.newSessionContext(ReferenceCountedOpenSslClientContext.java:149)
        at io.netty.handler.ssl.OpenSslClientContext.<init>(OpenSslClientContext.java:192)
        at io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:775)
        at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:446)
        at io.pravega.client.netty.impl.ConnectionFactoryImpl.establishConnection(ConnectionFactoryImpl.java:115)
        ... 11 common frames omitted
Caused by: java.lang.IllegalStateException: TrustManagerFactoryImpl is not initialized
        at sun.security.ssl.TrustManagerFactoryImpl.engineGetTrustManagers(TrustManagerFactoryImpl.java:100)
        at javax.net.ssl.TrustManagerFactory.getTrustManagers(TrustManagerFactory.java:285)
        at io.netty.handler.ssl.ReferenceCountedOpenSslClientContext.newSessionContext(ReferenceCountedOpenSslClientContext.java:133)
        ... 15 common frames omitted
```

**Problem location**

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by arvindkandhare**
Signed-off-by: arvindkandhare <arvind.kandhare@emc.com>

**Change log description**
In case of `standalone` clusters deployed with TLS enabled, the `autoscaleprocessor` client has `enablehostnamecheck`enabled. The default certificates provided with the standalone deployment can not have hostnames enabled.
Because of this the `AutoScaleProcessor` client should be created with hostname disabled.

**Purpose of the change**
Fixes #2580 

**What the code does**
Disable hostname checks for `AutoScaleProcessor`.

**How to verify it**
No stack trace is shown when `standalone` runs with TLS enabled.

### CODE REVIEW

**Comment by EronWright** [comment]
Have the security implications of this been discussed?  Generally speaking this is a bad practice.  Maybe a config setting (enabled by default) should be introduced.

**Comment by fpj** [comment]
Is there a way of doing it only for standalone?

**Comment by arvindkandhare** [comment]
> Please add a config setting for this.

Implemented a generic solution which collects the setting from config properties.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
     public static final Property<String> TLS_CERT_FILE = Property.named("tlsCertFile", "");
     public static final Property<Boolean> AUTH_ENABLED = Property.named("authEnabled", false);
     public static final Property<String> TOKEN_SIGNING_KEY = Property.named("tokenSigningKey", "secret");
+    public static final Property<Boolean> VALIDATE_HOSTNAME = Property.named("validateHostName", false);
```
Comment (EronWright) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java]: By the principle of secure-by-default, I would suggest that validation be enabled by default.

Comment (arvindkandhare) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java]: I have brought this up in earlier discussions as well. Here I go again :) - very few products in our ecosystem use (including Pravega) are secure by default.

Comment (arvindkandhare) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java]: > We are not on the same page here. I'm -1 on not being secure by default.

The default for `authEnabled` and `tlsEnabled` is `false`. Do you think that should be `true`? If yes, I will create another issue to fix this.

Comment (EronWright) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java]: My suggestion is simply that `validateHostName` be enabled by default, but that standalone's `tlsEnabled` be disabled by default.

**Pull request merged**